### ISSUE TITLE
db.Update duplication behavior

### ISSUE THREAD

**Issue by xJoeyv** [louischatriot/nedb#585]
My Document:
```
{"name":"Joey de Vries","account":"someid","password":"pass","admin":true,"rounds":[],"ig":{},"fb":{},"tw":{},"_id":"Rn0HqTbE6pAzig5I"}
```

When I now want to update IG to 
```
ig: {username: "pass"}}
```

It results in
```
{"name":"Joey de Vries","account":"someid","password":"pass","admin":true,"rounds":[],"ig":{},"fb":{},"tw":{},"_id":"Rn0HqTbE6pAzig5I"}
{"name":"Joey de Vries","account":"someid","password":"pass","admin":true,"rounds":[],"ig":{ username: "pass"},"fb":{},"tw":{},"_id":"Rn0HqTbE6pAzig5I"}
```
So it creates a new document and update there the data, but I want it to be in the same document else the file gets big very fast. Am I doing something wrong?

```
db.update(userDoc, { $set: { "ig": doc } }, {}, function (err, numReplaced) {
        if (numReplaced = 1) {
          resolve({ qcode: 1, data: {} });
        }
        resolve({ qcode: 0, message: "Unknown error please contact me@xjoeyv.com" });
 });
```
It is in a promise return.But this is how I update...

**Comment by JamesMGreene** [comment]
You can check out a similar discussion in #527, #530, etc., as well as reading through the ["Persistence" section of the README](https://github.com/louischatriot/nedb#persistence).

TL;DR: This is expected behavior until the datafile is compacted.  Compaction can occur in 3 different ways:
 1. When the database is loaded from file
 2. When you manually call the `db.persistence.compactDatafile()` method
 3. At regular intervals **if you schedule it** using `db.persistence.setAutocompactionInterval( 5000 /*ms*/ )`

Most importantly, please keep in mind that NeDB is **NOT** "creating multiple [records] with the same `_id`" so long as you are actually working with the data via NeDB's API.  The multiple associated lines that can be seen in the datafile are strictly representational in order to facilitate the append-only update implementation and can be ignored by standard consumers.

**Comment by kaustavsarkar** [comment]
I can confirm this behaviour. Although while updating it adds another object with updated values, however, adding Compaction deletes any duplicate records.
Adding to that, i think it is an overhead. Perhaps, I can check the code and see if it is feasible to get around it.

**Comment by digimbyte** [comment]
this is intended behavior, it works the same way in other systems like MongoDB. you just don't see it because it is encrypted. the files get compacted which remove older records, something that you have to call for your convenience as the DB is unusable while the data is being baked
